spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/TupleExpression.java

```diff
@@ -0,0 +1,54 @@
+package org.springframework.data.relational.core.sql;
+
+import org.jetbrains.annotations.NotNull;
+
+import static java.util.stream.Collectors.*;
+
+import java.util.List;
+
+/**
+ * A tuple as used in conditions like
+ * 
+ * <pre>
+ *   WHERE (one, two) IN (select x, y from some_table)
+ * </pre>
+ *
+ * @author Jens Schauder
+ * @since 3.5
+ */
+public class TupleExpression extends AbstractSegment implements Expression {
+
+	private final List<? extends Expression> expressions;
+
+	private static Segment[] children(List<? extends Expression> expressions) {
+		return expressions.toArray(new Segment[0]);
+	}
+
+	private TupleExpression(List<? extends Expression> expressions) {
+
+		super(children(expressions));
+
+		this.expressions = expressions;
+	}
+
+	public static TupleExpression create(Expression... expressions) {
+		return new TupleExpression(List.of(expressions));
+	}
+
+	public static TupleExpression create(List<? extends Expression> expressions) {
+		return new TupleExpression(expressions);
+	}
+
+	public static Expression maybeWrap(List<Column> columns) {
```

Comment:
How about moving this method into `Expressions` as `of(List<? extends Expression>)`? `maybeWrap` without documentation doesn't say anything about optimization into one expression. `Expressions.of(…)` could handle both cases with a clear and concise API turning multiple expressions into `TupleExpression`.

Reply:
Done